### ISSUE TITLE
package-directory-mismatch: do not report mismatch in the root directory

### ISSUE THREAD

**Issue by alexandear** [mgechev/revive#1539]
**Is your feature request related to a problem? Please describe.**

There are many libraries in the wild that have `.go` files in the project root directory with different package names. For example: https://gitlab.com/gitlab-org/api/client-go. The project directory is `client-go` while the package name is `gitlab`.

This issue only occurs when running Revive through the latest [Golangci-lint](https://github.com/golangci/golangci-lint/releases/tag/v2.5.0).

**Describe the solution you'd like**

The `package-directory-mismatch` rule should ignore issues when the directory is the project's root directory - e.g., when it contains `.git` or `go.mod`.

**Additional context**

To reproduce:

```
$ git clone https://gitlab.com/gitlab-org/api/client-go.git
$ cd client-go
$ golangci-lint run --config .golangci.yml --enable-only revive ./...
issues.go:17:9: package-directory-mismatch: package name "gitlab" does not match directory name "client-go" (revive)
package gitlab
        ^
keys_test.go:17:9: package-directory-mismatch: package name "gitlab" does not match directory name "client-go" (revive)
package gitlab
        ^
topics.go:17:9: package-directory-mismatch: package name "gitlab" does not match directory name "client-go" (revive)
package gitlab
        ^
3 issues:
* revive: 3
```

The contents of `.golangci.yml`:

```yml
version: "2"

linters:
  enable:
    - revive
  settings:
    revive:
      rules:
        - name: package-directory-mismatch
```

**Comment by alexandear** [comment]
I guess the issue in these lines:

https://github.com/mgechev/revive/blob/c603da318052d648cceb270a036c559bbbc8757c/rule/package_directory_mismatch.go#L99-L104

Maybe it's better to use `filepath.Abs` and then perform filtering.

**Comment by uloamaka** [comment]
Hi @alexandear I would like to submit a PR that resolves this.
